### DISCUSSION TITLE
Nginx Support (Contribution)

### DISCUSSION THREAD

**Discussion by bet0x** [mecha-cms/mecha#271] [Ideas]
### Make sure that your issue is reproducible by doing the following tasks

- [X] I am using application packages whose source code is not optimized for production (source code is not minified).
- [X] I have activated the test mode by changing the `TEST` constant value to `true` in the `.\index.php` file.
- [x] I have made sure that there are files named `error` or `error-x` or `error-y` in the `.\engine\log` folder.

### What installation method did you use?

Composer

### Describe your issue in the field below

Hello all,

I didn't found the documentation on git to place a PR but here it is the code block for Nginx to get it working and secured too:

    location / {
        try_files $uri $uri/ /index.php?$args;
    }

    location ~ \.php$ {
         fastcgi_intercept_errors on;
         include        fastcgi_params;
         fastcgi_param  SCRIPT_FILENAME  $document_root$fastcgi_script_name;
         fastcgi_pass unix:/run/php-fpm/www.sock;
    }

    location ~ /\. { deny all; }

    location ~* \.(?:ico|css|js|gif|jpe?g|png)$ {
        expires 30d;
        add_header Pragma public;
        add_header Cache-Control "public";
    }

### Package version

Development version (package on the `main` branch)

### Package version number

3.0.0

### What browsers are you seeing the problem on?

Firefox

**Comment by taufik-nurrohman** [comment]
Thanks a lot! Will add it to the site documentation 👍

**Comment by taufik-nurrohman** [comment]
@bet0x does this part prevent the root `.\index.php` file from public access?

~~~ nginxconf
location ~ \.php$ {
    fastcgi_intercept_errors on;
    include fastcgi_params;
    fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
    fastcgi_pass unix:/run/php-fpm/www.sock;
}
~~~

The root `.\index.php` must be accessible.

**Comment by bet0x** [comment]
Doesn't prevent access at all. It just works. I will apply a few fixes on other paths later but so far everything runs smoothly.

**Comment by taufik-nurrohman** [comment]
Okay, I will move this issue to the discussion later, so I can properly reference it. Discussions will also allow for more user inputs.

**Reply by bet0x** [reply]
Thank you @taufik-nurrohman.

**Comment by taufik-nurrohman** [comment]
Here&rsquo;s another alternative but quite complex: https://github.com/orgs/mecha-cms/discussions/108

**Comment by yuliakbard** [comment]
Hi @bet0x @taufik-nurrohman , what about the output image that is resized? My website, there is a problem with the image resizing output. Resized images produce a 404 error status

**Comment by taufik-nurrohman** [comment]
https://gist.github.com/taufik-nurrohman/1da313a2db48a26d836d781258b6c139
